cmd/validate/input.go

```diff
@@ -0,0 +1,258 @@
+
+			for _, f := range data.filePaths {
+				lock.Add(1)
+				go func(fpath string) {
+					defer lock.Done()
+
+					ctx := cmd.Context()
+					out, err := validate(ctx, fpath, data.policy, data.namespaces, data.effectiveTime, false)
+					res := result{
+						err: err,
+						input: input.Input{
+							FilePath: fpath,
+							Success:  err == nil,
+						},
+					}
+					// Skip on err to not panic. Error is return on routine completion.
+					if err == nil {
+						res.input.Violations = out.Violations()
+						showSuccesses, _ := cmd.Flags().GetBool("show-successes")
+						res.input.Warnings = out.Warnings()
+
+						successes := out.Successes()
+						res.input.SuccessCount = len(successes)
+						if showSuccesses {
+							res.input.Successes = successes
+						}
+						res.data = out.Data
+					}
+					res.input.Success = err == nil && len(res.input.Violations) == 0
+					ch <- res
+				}(f)
+			}
+
+			lock.Wait()
+			close(ch)
+
+			var inputs []input.Input
+			var manyData [][]evaluator.Data
+			var manyPolicyInput [][]byte
+			var allErrors error = nil
+
+			for r := range ch {
+				if r.err != nil {
+					e := fmt.Errorf("error validating file %s: %w", r.input.FilePath, r.err)
+					allErrors = multierror.Append(allErrors, e)
+				} else {
+					inputs = append(inputs, r.input)
+					manyData = append(manyData, r.data)
+					manyPolicyInput = append(manyPolicyInput, r.policyInput)
+				}
+			}
+			if allErrors != nil {
+				return allErrors
+			}
+
+			// Ensure some consistency in output.
+			sort.Slice(inputs, func(i, j int) bool {
+				return inputs[i].FilePath > inputs[j].FilePath
+			})
+
+			if len(data.outputFile) > 0 {
+				data.output = append(data.output, fmt.Sprintf("%s=%s", input.JSON, data.outputFile))
+			}
+
+			report, err := input.NewReport(inputs, data.policy, manyData, manyPolicyInput)
+			if err != nil {
+				return err
+			}
+
+			p := format.NewTargetParser(input.JSON, cmd.OutOrStdout(), utils.FS(cmd.Context()))
+			if err := report.WriteAll(data.output, p); err != nil {
+				return err
+			}
+
+			if data.strict && !report.Success {
+				// TODO: replace this with proper message and exit code 1.
+				return errors.New("success criteria not met")
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringSliceVarP(&data.filePaths, "file", "f", data.filePaths, "path to input YAML/JSON file (required)")
+
+	cmd.Flags().StringVarP(&data.policyConfiguration, "policy", "p", data.policyConfiguration, hd.Doc(`
+		Policy configuration as:
+		* Kubernetes reference ([<namespace>/]<name>)
+		* file (policy.yaml)
+		* git reference (github.com/user/repo//default?ref=main), or
+		* inline JSON ('{sources: {...}, configuration: {...}}')")`))
+
+	cmd.Flags().StringSliceVarP(&data.output, "output", "o", data.output, hd.Doc(`
+		write output to a file in a specific format, e.g. yaml=/tmp/output.yaml. Use empty string
+		path for stdout, e.g. yaml. May be used multiple times. Possible formats are json and yaml`))
```

Comment:
input/report.go appears to support `summary` in addition to `json` and `yaml`.

Reply:
Ack. Updated.